test/metabase/feature_extraction/async_test.clj

```diff
@@ -1,33 +1,31 @@
 (ns metabase.feature-extraction.async-test
   (:require [expectations :refer :all]
             [metabase.feature-extraction.async :refer :all]
-            [metabase.models.computation-job :refer [ComputationJob]]))
+            [metabase.models.computation-job :refer [ComputationJob]]
+            [metabase.test.async :refer [result!]]))
 
 (expect
   true
-  (let [job-id (compute (gensym) (constantly 1))]
-    (Thread/sleep 100)
+  (let [job-id (compute (gensym) (constantly 42))
+        _      (result! job-id)]
```

Comment:
same thing, the body of `let` is an implict `do` so why not just move `(result! job-id)` out of the bindings?